js/piwik.js

```diff
@@ -592,6 +592,43 @@ if (typeof window.Matomo !== 'object') {
 
             return matches ? matches[1] : url;
         }
+        function isPositiveNumberString(str) {
+            // !isNaN(str) could be used but does not cover '03' (octal) and '0xA' (hex)
+            // nor negative numbers
+            return (/^[1-9][0-9]*(\.[0-9]+)?$/).test(str);
+        }
+        function onlyNumbers(data) {
+            var result = {}, k;
+            for (k in data) {
+                if (data.hasOwnProperty(k)) {
+                    if (isNumber(data[k]) || isPositiveNumberString(data[k])) {
+                        result[k] = data[k];
+                    } else if (isDefined(data[k])) {
```

Comment:
just fyi because you iterate over `k` you could simplify this to `else {}` and things would still work even if `data[k] === undefined`. We might want to change it as otherwise maybe an error wouldn't be triggered when undefined is past? Haven't tested it though

Reply:
Not sure I understand what you mean here.

<img width="451" alt="image" src="https://user-images.githubusercontent.com/479499/132629722-1a000f6d-4e5e-4da0-8378-d0bc053b870a.png">